### PR TITLE
Clear device name error properly on provisioning

### PR DESCRIPTION
**Pull request by gabriel**
We weren't clearing device name error on input change.
This would cause the error to persist, even after changing the input text which was confusing to some users.

### CODE REVIEW

File: shared/login/register/paper-key/index.desktop.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ const styles = {
   },
   input: {
     marginBottom: 55,
-    height: '4em',
+    height: '5em',
```
Comment (gabriel) [inline shared/login/register/paper-key/index.desktop.js]: When testing provisioning and entering in paper key, my paper key was wrapping and causing scrollbar, bumping up the height, makes the input nicer and can see all of paper key entry.

**Comment by gabriel** [comment]
@keybase/react-hackers

**Comment by keybase-ci-visdiff** [comment]
The commits dd0d6c62723f...b4f69799868e introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Paper_Key-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498763256062-dd0d6c62723f-b4f69799868e-linux/Paper_Key-Normal-dd0d6c62723f.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498763256062-dd0d6c62723f-b4f69799868e-linux/Paper_Key-Normal-b4f69799868e.png)  
   Paper_Key-Normal comparison
</summary>

File: shared/login/register/set-public-name/container.js [hunk 2]
```diff
@@ -38,9 +38,23 @@ class _SetPublicName extends Component<void, ContainerProps, State> {
 
     this.state = {
       deviceName: null,
+      deviceNameError: null,
```
Comment (chrisnojima) [inline shared/login/register/set-public-name/container.js]: why isn't the prop just being updated in the store, why move this to local state?

**Comment by keybase-ci-visdiff** [comment]
The commits dd0d6c62723f...a09e5fa39bd8 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Paper_Key-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498766684130-dd0d6c62723f-a09e5fa39bd8-linux/Paper_Key-Normal-dd0d6c62723f.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498766684130-dd0d6c62723f-a09e5fa39bd8-linux/Paper_Key-Normal-a09e5fa39bd8.png)  
   Paper_Key-Normal comparison
</summary>

**Comment by chrisnojima** [comment]
i mean, if we're cleaing the error and its in the store, we should just update it in the store and this component is fine. why does it have its own cached version of the error? we should avoid having local state period

**Comment by gabriel** [comment]
Login was using a route prop and signup was using the store for deviceNameError.
I changed login to use the store like signup, and am triggering a clear of the error on input change.

**Comment by gabriel** [comment]
Route prop isn't a great place for something like an error cause it needs to be cleared on a input state change.

**Comment by keybase-ci-visdiff** [comment]
The commits 7f9625a81a54...2f2be2783002 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Paper_Key-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499289725516-7f9625a81a54-2f2be2783002-linux/Paper_Key-Normal-7f9625a81a54.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1499289725516-7f9625a81a54-2f2be2783002-linux/Paper_Key-Normal-2f2be2783002.png)  
   Paper_Key-Normal comparison
</summary>

**Pull request merged**